contracts/PodCastNFT.sol

```diff
@@ -260,11 +260,55 @@ contract PodCastNFT is ERC721, Ownable {
             "You cannot answer twice"
         );
         userAttribute[msg.sender].point += 100;
+        string memory podcastRole = checkPodcastRole();
+        if (bytes(podcastRole).length != bytes("").length) {
+            roles[msg.sender].push(podcastRole);
+        }
         userAttribute[msg.sender].claimableToken += 100e18;
         userAttribute[msg.sender].answeredAt = block.timestamp;
         emit CheckedAnswer(msg.sender, userAttribute[msg.sender].answeredAt);
     }
 
+    function checkPodcastRole() private returns (string memory) {
+        uint256 podcastPoint = userAttribute[msg.sender].point;
+        if (
+            podcastPoint >= 20000 &&
+            hasRoleOf(msg.sender, "Podcast God") == false
+        ) {
+            return "Podcast God";
+        } else if (
+            podcastPoint >= 10000 &&
+            hasRoleOf(msg.sender, "Podcast Emperor") == false
+        ) {
+            return "Podcast Emperor";
+        } else if (
+            podcastPoint >= 5000 &&
+            hasRoleOf(msg.sender, "Podcast King") == false
+        ) {
+            return "Podcast King";
+        } else if (
+            podcastPoint >= 3000 &&
+            hasRoleOf(msg.sender, "Podcast Lord") == false
+        ) {
+            return "Podcast Lord";
+        } else if (
+            podcastPoint >= 1000 &&
+            hasRoleOf(msg.sender, "Podcast Master") == false
+        ) {
+            return "Podcast Master";
+        } else if (
+            podcastPoint >= 500 &&
+            hasRoleOf(msg.sender, "Podcast Veteran") == false
+        ) {
+            return "Podcast Veteran";
+        } else {
+            if (hasRoleOf(msg.sender, "Podcast Contributor") == false) {
+                return "Podcast Contributor";
+            }
+        }
+        return "";
```

Comment:
I think it's supposed to come here when `hasRoleOf(msg.sender, "Podcast Contributor") == true` , but personally I’d rather determine the above case at L:306, and L:309 returns `"Podcast Contributor"` for the first time podcast contribution.

Reply:
I couldn't get an image, what kind of code are you imagining?